browser/extensions/api/binance_api.cc

```diff
@@ -39,5 +41,22 @@ BinanceGetUserTLDFunction::Run() {
       std::make_unique<base::Value>(user_TLD)));
 }
 
+ExtensionFunction::ResponseAction
+BinanceIsSupportedRegionFunction::Run () {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  if (brave::IsTorProfile(profile)) {
+    return RespondNow(Error("Not available in Tor profile"));
+  }
+
+  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile);
+  if (!ads_service) {
+    return RespondNow(Error("Ads Service unavailable."));
+  }
+
+  const bool is_supported = ads_service->IsSupportedLocale();
```

Comment:
Could we refactor this code and the ads code to use a common function? Rather than including an ads dependency here. In case for example ads is disabled.  We'd have to for example at a minimum have `#if BUILDFLAG(BRAVE_ADS_ENABLED)` but it's better to just not have any ads code here.

Reply:
As discussed this has been updated to use our own blacklist, as opposed to the ads whitelist